scene/gui/dialogs.cpp

```diff
@@ -337,11 +348,11 @@ void AcceptDialog::remove_button(Control *p_button) {
 		button->disconnect("pressed", callable_mp(this, &AcceptDialog::_cancel_pressed));
 	}
 
-	Node *right_spacer = buttons_hbox->get_child(button->get_index() + 1);
-	// Should always be valid but let's avoid crashing.
+	Control *right_spacer = Object::cast_to<Control>(button->get_meta("__right_spacer"));
```

Comment:
I think you can still move it above the disconnects and add something like this, similar to the check for the button:

```
ERR_FAIL_COND_MSG(right_spacer->get_parent() != buttons_hbox, vformat("Cannot remove button %s as its associated spacer does not belong to this dialog.", button->get_name()));
```